### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by anatolii-zarytskyi**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert cinemaHall " + cinemaHall, e);
```
Comment (iedenysenko) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:29]:
```suggestion
            throw new DataProcessingException("Can't insert cinema hall " + cinemaHall, e);
```

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            throw new DataProcessingException("Can't get a cinema hall by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM CinemaHall", CinemaHall.class).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get cinema halls from DB ", e);
```
Comment (iedenysenko) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:51]:
```suggestion
            throw new DataProcessingException("Can't get all cinema halls from DB ", e);
```

File: src/main/java/mate/academy/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -45,6 +45,10 @@ public Movie add(Movie movie) {
 
     @Override
     public List<Movie> getAll() {
-        return null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Movie", Movie.class).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get list of movies from DB ", e);
```
Comment (iedenysenko) [inline src/main/java/mate/academy/dao/impl/MovieDaoImpl.java:51]:
```suggestion
            throw new DataProcessingException("Can't get list of all movies from DB ", e);
```

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM MovieSession "
+                    + "WHERE movie.id =:movieId "
+                    + "AND showTime BETWEEN :start AND :end", MovieSession.class)
+                    .setParameter("movieId", movieId)
+                    .setParameter("start", date.atStartOfDay())
+                    .setParameter("end", date.atStartOfDay().plusDays(1L))
```
Comment (iedenysenko) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:55]:
```suggestion
                    .setParameter("end", date.atTime(LocalTime.MAX))
```

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            throw new DataProcessingException("Can't get a movie session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM MovieSession "
+                    + "WHERE movie.id =:movieId "
+                    + "AND showTime BETWEEN :start AND :end", MovieSession.class)
```
Comment (iedenysenko) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:52]: I think you need to use join fetch...

File: src/main/java/mate/academy/Main.java [hunk 6]
```diff
@@ -1,16 +1,41 @@
...
+    private static final CinemaHallService cinemaHallService =
+            (CinemaHallService) injector.getInstance(CinemaHallService.class);
+    private static final MovieSessionService movieSessionService =
+            (MovieSessionService) injector.getInstance(MovieSessionService.class);
 
+    public static void main(String[] args) {
         Movie fastAndFurious = new Movie("Fast and Furious");
         fastAndFurious.setDescription("An action film about street racing, heists, and spies.");
         movieService.add(fastAndFurious);
         System.out.println(movieService.get(fastAndFurious.getId()));
```
Comment (iedenysenko) [inline src/main/java/mate/academy/Main.java:26]: I think you need to implement the method getAll in MovieDaoImpl

**Review by iedenysenko — CHANGES_REQUESTED** [review]

**Review by iedenysenko — APPROVED** [review]

**Review by Resci — APPROVED** [review]

**Pull request closed without merging**